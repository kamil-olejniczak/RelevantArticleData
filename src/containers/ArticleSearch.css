.ArticleSearch {
    width: 70%;
    margin: 0 auto;
}

.ArticleSearch input {
    box-sizing: border-box;
    width: 100%;
    padding: 20px 20px;
    margin-bottom: 30px;
    border: none;
    -webkit-border-radius: 5px;
    -moz-border-radius: 5px;
    border-radius: 5px;
    font-size: 1.2em;
}

.ArticleSearch div h1, h2 {
    text-align: center;
    color: #7c795d;
}

.ArticleSearch h1 {
    font-size: 3em;
}

.ArticleSearch h2 {
    font-size: 2em;
}