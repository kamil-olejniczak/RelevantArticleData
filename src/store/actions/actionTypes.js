export const INIT_GET_ARTICLE = 'INIT_GET_ARTICLE';
export const RESOLVING_ARTICLE = 'RESOLVING_ARTICLE';
export const GET_ARTICLE = 'GET_ARTICLE';
export const CLEAR_ARTICLE = 'CLEAR_ARTICLE';
export const INIT_RELOAD_ARTICLE = 'INIT_RELOAD_ARTICLE';
export const RELOAD_ARTICLE = 'RELOAD_ARTICLE';
export const INIT_CLEAR_DATA = 'INIT_CLEAR_DATA';
export const CLEAR_DATA = 'CLEAR_DATA';
export const INIT_SAVE_DATA = 'INIT_SAVE_DATA';
export const SAVE_DATA = 'SAVE_DATA';

export const INVALID_URL = 'INVALID_URL';
export const CLOSE_MODAL = 'CLOSE_MODAL';

export const ERROR_OCCURRED = 'ERROR_OCCURRED';
