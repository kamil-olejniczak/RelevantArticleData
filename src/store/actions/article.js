import * as actionTypes from './actionTypes';

export const articleResponse = ({data, content, recentlyVisited, statistics}, type) => ({
    type,
    data,
    content,
    recentlyVisited: recentlyVisited,
    statistics
});

export const getSpecifiedArticle = (url) => ({
    type: actionTypes.INIT_GET_ARTICLE,
    url
});

export const resolvingArticle = () => ({
    type: actionTypes.RESOLVING_ARTICLE
});

export const clearArticle = () => ({
    type: actionTypes.CLEAR_ARTICLE,
});

export const reloadPreviouslyVisitedArticle = (url) => ({
    type: actionTypes.INIT_RELOAD_ARTICLE,
    url
});

export const invalidUrl = () => ({
    type: actionTypes.INVALID_URL,
});

export const closeModalWithData = () => ({
    type: actionTypes.CLOSE_MODAL,
});

export const errorOccurred = (error) => ({
    type: actionTypes.ERROR_OCCURRED,
    error
});
