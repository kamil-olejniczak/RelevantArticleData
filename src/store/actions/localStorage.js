import * as actionTypes from './actionTypes';

export const clearData = () => ({
    type: actionTypes.INIT_CLEAR_DATA,
});

export const localStorageWasCleaned = () => ({
    type: actionTypes.CLEAR_DATA,
});

export const saveData = formattedData => ({
    type: actionTypes.INIT_SAVE_DATA,
    formattedData
});

export const localStorageWasUpdated = () => ({
    type: actionTypes.SAVE_DATA,
});