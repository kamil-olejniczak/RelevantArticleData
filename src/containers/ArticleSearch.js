import React, {Component} from 'react';
import {connect} from 'react-redux';
import * as articleActions from "../store/actions/article";
import './ArticleSearch.css';

class ArticleSearch extends Component {
    searchForArticle = (event) => {
        if (event.key === 'Enter') {
            this.props.onGetSpecifiedArticle(event.target.value);
        }
    };

    render() {
        return (
            !this.props.isArticleShown ? (
                <div className="ArticleSearch">
                    <div>
                        <h1>Relevant Article Data</h1>
                        <h2>Read only valuable content</h2>
                    </div>
                    <input className={this.props.error ? "reducedBottomMargin" : null}
                           type="text"
                           onKeyDown={this.searchForArticle}
                           placeholder="Paste link to article..."/>
                    {this.props.error ? (<div className="warning">{this.props.error}</div>) : null}
                </div>
            ) : null
        );
    }
}

const mapDispatchToProps = dispatch => {
    return {
        onGetSpecifiedArticle: (url) => dispatch(articleActions.getSpecifiedArticle(url))
    };
};

const mapStateToProps = state => {
    return {
        error: state.article.error,
        isArticleShown: state.article.isArticleShown
    }
};

export default connect(mapStateToProps, mapDispatchToProps)(ArticleSearch);