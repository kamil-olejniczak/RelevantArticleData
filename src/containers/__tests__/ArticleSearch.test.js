import * as React from "react";
import {shallow} from 'enzyme';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import ArticleSearch from '../ArticleSearch';
import * as actionTypes from "../../store/actions/actionTypes";

describe('ArticleSearch tests', () => {
    const mockStore = configureStore();

    let wrapper, store;

    it('renders correctly', () => {
        store = mockStore(mockStoreWithoutErrorAndArticle);
        const component = renderer.create(<ArticleSearch store={store}/>).toJSON();

        expect(component).toMatchSnapshot();
    });

    it('should not render component when article is shown', () => {
        store = mockStore(mockStoreWithVisibleArticle);
        wrapper = shallow(<ArticleSearch store={store}/>).dive();

        expect(wrapper.find('.ArticleSearch')).toHaveLength(0);
    });

    it('should render component when article is not shown', () => {
        store = mockStore(mockStoreWithoutErrorAndArticle);
        wrapper = shallow(<ArticleSearch store={store}/>).dive();

        expect(wrapper.find('.ArticleSearch')).toHaveLength(1);
    });

    it('input should not have any class when error is false', () => {
        store = mockStore(mockStoreWithoutErrorAndArticle);
        wrapper = shallow(<ArticleSearch store={store}/>).dive();

        expect(wrapper.find('input').hasClass('reducedBottomMargin')).toBeFalsy();
    });

    it('input should have reducedBottomMargin when error is true', () => {
        store = mockStore(mockStoreWithError);
        wrapper = shallow(<ArticleSearch store={store}/>).dive();

        expect(wrapper.find('input').hasClass('reducedBottomMargin')).toBeTruthy();
    });

    it('should not render div with warning when error is false', () => {
        store = mockStore(mockStoreWithoutErrorAndArticle);
        wrapper = shallow(<ArticleSearch store={store}/>).dive();

        expect(wrapper.find('.warning')).toHaveLength(0);
    });

    it('should render div with warning when error is true', () => {
        store = mockStore(mockStoreWithError);
        wrapper = shallow(<ArticleSearch store={store}/>).dive();

        expect(wrapper.find('.warning')).toHaveLength(1);
    });


    it('should dispatch INIT_GET_ARTICLE action from searchForArticle method', () => {
        const fakeData = 'fakeValue';
        store = mockStore(mockStoreWithError);
        wrapper = shallow(<ArticleSearch store={store}/>).dive();

        wrapper.find('input').simulate('keydown', {key: 'Enter', target: {value: fakeData}});

        expect(store.getActions()).toEqual([{type: actionTypes.INIT_GET_ARTICLE, url: fakeData}]);
    });
});

const mockStoreWithVisibleArticle = {article: {isArticleShown: true, error: false}};
const mockStoreWithError = {article: {isArticleShown: false, error: true}};
const mockStoreWithoutErrorAndArticle = {article: {isArticleShown: false, error: false}};