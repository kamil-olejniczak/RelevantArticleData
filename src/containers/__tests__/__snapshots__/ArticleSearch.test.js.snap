// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleSearch tests renders correctly 1`] = `
<div
  className="ArticleSearch"
>
  <div>
    <h1>
      Relevant Article Data
    </h1>
    <h2>
      Read only valuable content
    </h2>
  </div>
  <input
    className={null}
    onKeyDown={[Function]}
    placeholder="Paste link to article..."
    type="text"
  />
</div>
`;
